### ISSUE TITLE
Slow import of data volumes with many restarts of the importer pod

### ISSUE THREAD

**Issue by fungiboletus** [kubevirt/containerized-data-importer#2182]
/kind bug

**What happened**:

Creating a virtual machine using a data volume and a cloud image from http is very slow.

**What you expected to happen**:

The server can download the image and create the qemu disk storage in less than a minute if done manually, but it takes many minutes using kubevirt and cdi.

**How to reproduce it (as minimally and precisely as possible)**:

I have the following vm file:
```yaml
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: test-vm
  namespace: default
  labels:
    kubevirt.io/vm: test-vm
spec:
  running: true
  template:
    metadata:
      labels:
        kubevirt.io/vm: test-vm
        kubevirt.io/size: medium
        kubevirt.io/domain: test
    spec:
      domain:
        cpu:
          cores: 6
        devices:
          disks:
            - name: disk0
              disk:
                bus: sata
            - name: cloudinitdisk
              disk:
                bus: virtio
          interfaces:
            - name: default
              masquerade: {}
        resources:
          requests:
            memory: 8192M
      networks:
        - name: default
          pod: {}
      volumes:
        - name: disk0
          dataVolume:
            name: ubuntu-focal-datavolume
        - name: cloudinitdisk
          cloudInitNoCloud:
            userData: |
              #cloud-config
              ssh_authorized_keys:
                - ssh-ed25519 mykey user@laptop
              users:
                - default
  dataVolumeTemplates:
    - metadata:
        name: ubuntu-focal-datavolume
      spec:
        pvc:
          storageClassName: local-path
          accessModes:
            - ReadWriteOnce
          resources:
            requests:
              storage: 32Gi
        source:
          http:
            url: https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img 
```

The importer pods is created immediately but takes a long time, more than 10 minutes and restarts a few times (like 4) during the process. It runs faster if I use less storage, the example in the lab uses 5Gi for Fedora and it is faster but still not fast.

**Anything else we need to know?**:

I also tried longhorn instead of local-path from K3S for the storage class, but the behaviour is the same. 

Also, once up and running the VM is as fast as expected and doesn't seem to suffer from IO performances.

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`): `v1.38.1`
- Kubernetes version (use `kubectl version`): `v1.22.6+k3s1`
- Cloud provider or hardware configuration: Hetzner bare metal dedicated server running Ubuntu Server and K3S. Gen3 AMD Ryzen CPU and fast local NVMe SSD.
- Install tools: K3S
- Others: /

**Comment by fungiboletus** [comment]
After testing a bit more, I noticed that cloud-images.ubuntu.com blocks my server now. The ubuntu download server may have a very agressive rate-limit thing or CDI may make a lot of network requests.

```
could not convert image to raw Log line from nbdkit: nbdkit: curl[2]: error: problem doing HEAD request to fetch size of URL [https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img]: Timeout was reached: Failed to connect to cloud-images.ubuntu.com port 443: Connection timed out
```

So I uploaded the ubuntu image somewhere else (azure blob storage in my case). The import did run fast this time, which corresponds to the network speed and the ssd speeds plus some overhead.

Could the slowness be caused by ratelimits on public cloud images download servers ?
